bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/internal/GC100IRActiveBinding.java

```diff
@@ -0,0 +1,338 @@
+
+	/**
+	 * the refresh interval which is used to check for lost connections
+	 * (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	public void activate() {
+		logger.debug(getName() + " activate()");
+		setProperlyConfigured(true);
+	}
+
+	public void deactivate() {
+		logger.debug(getName() + " deactivate()");
+
+		// close any open connections
+		for (GC100IRConnector connector : connectors.values()) {
+			if (connector.isConnected()) {
+				connector.close();
+			}
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "GC100IR Refresh Service";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void bindingChanged(BindingProvider provider, String itemName) {
+		if (provider instanceof GC100IRBindingProvider) {
+			GC100IRBindingProvider gc100Provider = (GC100IRBindingProvider) provider;
+			registerWatch(gc100Provider, itemName);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void allBindingsChanged(BindingProvider provider) {
+		if (provider instanceof GC100IRBindingProvider) {
+			GC100IRBindingProvider gc100Provider = (GC100IRBindingProvider) provider;
+			for (String itemName : gc100Provider.getItemNames()) {
+				registerWatch(gc100Provider, itemName);
+			}
+		}
+	}
+	/**
+	 * Registers all items and GC100IRConnector.
+	 */
+	private void registerAllWatches() {
+		for (BindingProvider provider : providers) {
+			if (provider instanceof GC100IRBindingProvider) {
+				GC100IRBindingProvider gc100Provider = (GC100IRBindingProvider) provider;
+				for (String itemName : gc100Provider.getItemNames()) {
+					registerWatch(gc100Provider, itemName);
+				}
+			}
+		}
+	}
+	/**
+	 * Register each item and adds them in ItemUtility list.
+	 * 
+	 * @param gc100Provider
+	 * @param itemName
+	 */
+	private void registerWatch(GC100IRBindingProvider gc100Provider,
+			String itemName) {
+
+		if (!gc100Provider.providesBindingFor(itemName)) {
+			return;
+		}
+
+		String gc100Instance = gc100Provider.getGC100Instance(itemName);
+		int module = gc100Provider.getGC100Module(itemName);
+		int connector = gc100Provider.getGC100Connector(itemName);
+		String code = gc100Provider.getCode(itemName);
+
+		GC100ItemBean itemBean = new GC100ItemBean();
+		itemBean.setGC100Instance(gc100Instance);
+		itemBean.setModule(module);
+		itemBean.setConnector(connector);
+		itemBean.setCode(code);
+
+		ItemUtility itemUtility = ItemUtility.getInstance();
+		itemUtility.addItem(itemName, itemBean);
+
+		GC100IRConnector gc100Connector = getGC100Connector(gc100Instance);
+
+		if (gc100Connector == null) {
+			logger.error("Connection failed with " + gc100Instance);
+		}
+	}
+	/**
+	 * Gets the GC100Instance according to item name.
+	 * 
+	 * @param itemName
+	 * @return a String value of GC100 instance.
+	 */
+	private String getGC100Instance(String itemName) {
```

Comment:
please rename to something like "getGC100InstanceName" since no instance is being returned here